framework/src/interfaces/DataFileInterface.C

```diff
@@ -0,0 +1,84 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "DataFileInterface.h"
+#include "MooseError.h"
+
+template <class T>
+DataFileInterface<T>::DataFileInterface(const T & parent) : _parent(parent)
+{
+}
+
+template <class T>
+std::string
+DataFileInterface<T>::getDataFileName(const std::string & param) const
+{
+  /// - relative to the input file directory
+  {
```

Comment:
Codes from Line 22 to Line 30 except the line 24 should be moved into `getDataFileNameByName` at its beginning imo. Otherwise, calling `getDataFileNameByName` will skip the relative path part.

Reply:
That would mean refactoring the code that turns a relative path into an absolute path applied to the input file location (this is done in handling of the `FileName` parameter, which is why it is only in the first method that has access to that parmeter.

Reply:
Where that `turns a relative path into an absolution path` code is at? Maybe we can bypass `getParam` and directly call that code. I did not see the code so not sure if it is possible. But the code here can create a pitfall in my opinion and need to be fixed.